qt/scientific_interfaces/Indirect/IndirectSymmetriseView.cpp

```diff
@@ -0,0 +1,434 @@
+  auto positiveERaw = m_uiForm.ppRawPlot->getRangeSelector("PositiveE");
+
+  value = fabs(value);
+
+  if (prop == m_properties["EMin"]) {
+    positiveERaw->setMinimum(value);
+  }
+
+  if (prop == m_properties["EMax"]) {
+    positiveERaw->setMaximum(value);
+  }
+}
+
+/**
+ * Handles the X minimum value being changed from a range selector.
+ *
+ * @param value New range selector value
+ */
+void IndirectSymmetriseView::xRangeMinChanged(double value) {
+  m_dblManager->setValue(m_properties["EMin"], std::abs(value));
+  m_uiForm.pbPreview->setEnabled(true);
+}
+
+/**
+ * Handles the X maximum value being changed from a range selector.
+ *
+ * @param value New range selector value
+ */
+void IndirectSymmetriseView::xRangeMaxChanged(double value) {
+  m_dblManager->setValue(m_properties["EMax"], std::abs(value));
+  m_uiForm.pbPreview->setEnabled(true);
+}
+
+void IndirectSymmetriseView::updateRunButton(bool enabled, std::string const &enableOutputButtons,
+                                             QString const &message, QString const &tooltip) {
+  setRunEnabled(enabled);
+  m_uiForm.pbRun->setText(message);
+  m_uiForm.pbRun->setToolTip(tooltip);
+  if (enableOutputButtons != "unchanged")
+    setSaveEnabled(enableOutputButtons == "enable");
+}
+
+void IndirectSymmetriseView::setRunEnabled(bool enabled) { m_uiForm.pbRun->setEnabled(enabled); }
+
+void IndirectSymmetriseView::setSaveEnabled(bool enabled) { m_uiForm.pbSave->setEnabled(enabled); }
+
+void IndirectSymmetriseView::setFBSuffixes(QStringList const suffix) { m_uiForm.dsInput->setFBSuffixes(suffix); }
+
+void IndirectSymmetriseView::setWSSuffixes(QStringList const suffix) { m_uiForm.dsInput->setWSSuffixes(suffix); }
+
+/**
+ * Plots a new workspace in the mini plot when it is loaded form the data
+ *selector.
+ *
+ * @param workspaceName Name of the workspace that has been loaded
+ */
+void IndirectSymmetriseView::plotNewData(QString const &workspaceName) {
+  // Set the preview spectrum number to the first spectrum in the workspace
+  auto sampleWS = AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>(workspaceName.toStdString());
+  int minSpectrumRange = sampleWS->getSpectrum(0).getSpectrumNo();
+  m_dblManager->setValue(m_properties["PreviewSpec"], static_cast<double>(minSpectrumRange));
+
+  updateMiniPlots();
+
+  // Set the preview range to the maximum absolute X value
+  auto const axisRange = getXRangeFromWorkspace(sampleWS);
+  double symmRange = std::max(fabs(axisRange.first), fabs(axisRange.second));
+
+  // Set valid range for range selectors
+  auto positiveESelector = m_uiForm.ppRawPlot->getRangeSelector("PositiveE");
+  positiveESelector->setBounds(axisRange.first, axisRange.second);
+  positiveESelector->setRange(0, symmRange);
+
+  // Set some default (and valid) values for E range
+  m_dblManager->setValue(m_properties["EMax"], axisRange.second);
+  m_dblManager->setValue(m_properties["EMin"], axisRange.second / 10);
```

Comment:
why does this one get a divide by 10 and the EMax does not?

Reply:
because this way the range markers will be set to be between max/10 and max. this will always be a valid setting.

Reply:
can we have  a comment?

Reply:
is it not clear from the `Set some default (and valid) values for E range` comment?

Reply:
it was more the logic for the 10. Can just add something like "divide by 10 to ensure that the min and max are always valid"